### PR TITLE
Check that the base CircleCI configuration matches the user's too

### PR DESCRIPTION
**Pull request by ivomurrell**
# Description

We were mistakenly only checking that the config returned by each hook was a subset of the user's CircleCI config, but we additionally want to check that the base configuration included regardless of hooks also matches. This is so that we check that fields like the Tool Kit orb version or the node executor are correct.

# Checklist:

- [x] My branch has been rebased onto the latest commit on main (don't merge main into your branch)
- [x] My commit messages are [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), for example: `feat(circleci): add support for nightly workflows`, `fix: set Heroku app name for staging apps too`

### CODE REVIEW

File: plugins/circleci/src/circleci-config.ts [hunk 1]
```diff
@@ -196,8 +198,10 @@ export default abstract class CircleCiConfigHook extends Hook<CircleCIState> {
   _versionTag?: string
   abstract config: CircleCIStatePartial
 
-  async getCircleConfig(): Promise<string | undefined> {
+  async getCircleConfig(): Promise<[string | undefined, boolean]> {
+    let firstLoad = false
```
Comment (joelcarr) [inline plugins/circleci/src/circleci-config.ts:202]: **nitpick:** This to me reads as _this is not the first load_ whereas we want to say _the first load hasn't happened yet_. To me I would name this `hasCheckedBaseConfig` to be super verbose matching the intent of this Boolean.

File: plugins/circleci/test/circleci-config.test.ts [hunk 2]
```diff
@@ -18,10 +18,19 @@ jest.mock('fs', () => {
...
 
 describe('CircleCI config hook', () => {
-  class TestHook extends CircleCiConfigHook {
+  abstract class FakeCircleCiConfigHook extends CircleCiConfigHook {
+    async getCircleConfig(): Promise<[string | undefined, boolean]> {
+      const [config] = await super.getCircleConfig()
+      // pretend it's never the first time we've opened the config so that we
+      // skip the comparison with the base config generated by
+      // getInitialState()
+      return [config, false]
```
Comment (joelcarr) [inline plugins/circleci/test/circleci-config.test.ts:27]: **chore:** We should add a test for when it has been the config has been checked once before.

File: plugins/circleci/src/circleci-config.ts (hunk 1 above)
Comment (ivomurrell) [reply plugins/circleci/src/circleci-config.ts:202]: I'm confused, it _does_ mean "this is not the first load", which is how you read it. Are you saying you want it to mean "the first load hasn't happened yet"?

Comment (joelcarr) [reply plugins/circleci/src/circleci-config.ts:202]: Yes 👍 To me - as soon as `getCircleConfig()` is called, that is the first load so `firstLoad` should be true. I was not able to figure out the purpose/intent of `firstLoad` without parsing the code in my head. My aim for the nitpick is to make the variable name clearer so I can now its' intent without having to read the whole function and how it is used.

Comment (ivomurrell) [reply plugins/circleci/src/circleci-config.ts:202]: I'd argue that we shouldn't change the variable to something like `hasCheckedBaseConfig` as `getCircleConfig` shouldn't concern itself with what callees are doing with the `firstLoad` (or whatever) boolean – it's a leaky abstraction, and might no longer be correctly named if we later add another piece of logic that uses the boolean.

Comment (joelcarr) [reply plugins/circleci/src/circleci-config.ts:202]: Is there anyway we can make this clearer? I struggled to understand its' intent when first reading the block of code. The aim here is to make this easier to maintain.

Comment (ivomurrell) [reply plugins/circleci/src/circleci-config.ts:202]: `isFirstLoad`? `isFirstLoadOfConfig`? I'm still struggling to understand what's unclear about it, sorry!

> as soon as getCircleConfig() is called, that is the first load so firstLoad should be true

this isn't true the majority of times it's called, only the first time it's called 😅

**Pull request merged**